llvm/tools/llvm-c-test/atomic.c

```diff
@@ -0,0 +1,64 @@
+/*===-- atomic.c - tool for testing libLLVM and llvm-c API ----------------===*\
+|*                                                                            *|
+|* Part of the LLVM Project, under the Apache License v2.0 with LLVM          *|
+|* Exceptions.                                                                *|
+|* See https://llvm.org/LICENSE.txt for license information.                  *|
+|* SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception                    *|
+|*                                                                            *|
+|*===----------------------------------------------------------------------===*|
+|*                                                                            *|
+|* This file implements the --atomic-* commands in llvm-c-test.               *|
```

Comment:
This is not going to do anything if you don't also add a lit test calling `llvm-c-test --atomic-syncscopre`.

I think it would be preferable to add this as a unit test in llvm/unittest/IR/InstructionsTest.cpp instead.